tests/test_squirrel.c

```diff
@@ -66,25 +69,27 @@ Test(squirrel_unit,sqr_lowest4_vals_to_front){
 
   sqr_lowest4_vals_to_front(x,y,5,2);
 
-  cr_assert_float_eq(x[0], -133.01, 1e-3, "best fitness should be 3.0");
-  cr_assert_float_eq(x[1], -133.0, 1e-3, "best fitness replaced by 3rd element");
-  cr_assert_float_eq(x[2], 3.0, 1e-3, "best fitness should be 3.0");
-  cr_assert_float_eq(x[3], 12.5, 1e-3, "best fitness replaced by 3rd element");
-  cr_assert_float_eq(x[4], 133.0, 1e-3, "best fitness replaced by 3rd element");
-
-  cr_assert_float_eq(y[0], 4, 1e-3, "best fitness at 4,4");
-  cr_assert_float_eq(y[1], 4, 1e-3, "best fitness at 4,4");
-  cr_assert_float_eq(y[2], 3, 1e-3, "2nd best fitness at 3,3");
-  cr_assert_float_eq(y[3], 3, 1e-3, "2nd best fitness at 3,3");
-  cr_assert_float_eq(y[4], 2, 1e-3, "3rd best fitness at 2,2");
-  cr_assert_float_eq(y[5], 2, 1e-3, "3rd best fitness at 2,2");
-  cr_assert_float_eq(y[6], 1, 1e-3, "4th best fitness at 1,1");
-  cr_assert_float_eq(y[7], 1, 1e-3, "4th best fitness at 1,0");
-  cr_assert_float_eq(y[8], 0, 1e-3, "worst best fitness at 0,0");
-  cr_assert_float_eq(y[9], 0, 1e-3, "worst best fitness at 0,0");
+  cr_expect_float_eq(x[0], -133.01, 1e-3, "best fitness should be 3.0");

```

Comment:
@abhimanyu232 please check why using `FLT_EPSILON` breaks this test. Why is there a precision loss here?